File: Examples/CSharp/Images/BuildDocAndInsertImageUsingImageStream.cs
using System.IO;
using System;

namespace Aspose.Note.Examples.CSharp.Images
{
    public class BuildDocAndInsertImageUsingImageStream
    {
        public static void Run()
        {
            // ExStart:BuildDocAndInsertImageUsingImageStream
            // ExFor:Document
            // ExFor:Image
            // ExFor:Image.Alignment
            // ExSummary:Shows how to add an image from stream to a document.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_Images();

            // Create an object of the Document class
            Document doc = new Document();

            // Initialize Page class object
            Page page = new Page();

            Outline outline1 = new Outline();
            OutlineElement outlineElem1 = new OutlineElement();

            using (FileStream fs = File.OpenRead(dataDir + "image.jpg"))
            {

                // Load the second image using the image name, extension and stream.
                Image image1 = new Image("Penguins.jpg", fs)
                                               {
                                                   // Set image alignment
                                                   Alignment = HorizontalAlignment.Right
                                               };

                outlineElem1.AppendChildLast(image1);
            }

            outline1.AppendChildLast(outlineElem1);
            page.AppendChildLast(outline1);

            doc.AppendChildLast(page);

            // Save OneNote document
            dataDir = dataDir + "BuildDocAndInsertImageUsingImageStream_out.one";
            doc.Save(dataDir);

            // ExEnd:BuildDocAndInsertImageUsingImageStream 
            
            Console.WriteLine("\nDocument created and image using image stream inserted successfully.\nFile saved at " + dataDir); 
        }
    }
}

File: Examples/VisualBasic/Images/BuildDocAndInsertImageUsingImageStream.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Imports System.Drawing
Namespace Images
    Public Class BuildDocAndInsertImageUsingImageStream
        Public Shared Sub Run()
            ' ExStart:BuildDocAndInsertImageUsingImageStream
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_Images()

            ' Create an object of the Document class
            Dim doc As New Document()
            ' Initialize Page class object
            Dim page As New Aspose.Note.Page(doc)

            Dim outline1 As New Outline(doc) With {
                 .VerticalOffset = 600,
                 .HorizontalOffset = 0
            }
            Dim outlineElem1 As New OutlineElement(doc)
            Dim fs As FileStream = File.OpenRead(dataDir & Convert.ToString("image.jpg"))
            ' Load the second image using the image name, extension and stream.
            Dim image1 As New Aspose.Note.Image(doc, "Penguins", "jpg", fs)
            ' Set image alignment
            image1.Alignment = HorizontalAlignment.Right

            outlineElem1.AppendChild(image1)
            outline1.AppendChild(outlineElem1)
            page.AppendChild(outline1)

            doc.AppendChild(page)

            dataDir = dataDir & Convert.ToString("BuildDocAndInsertImageUsingImageStream_out_.one")
            ' Save OneNote document
            doc.Save(dataDir)
            ' ExEnd:BuildDocAndInsertImageUsingImageStream 
            Console.WriteLine(Convert.ToString(vbLf & "Document created and image using image stream inserted successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace